### ISSUE TITLE
Don't store FabricInfo without usable operational keys

### ISSUE THREAD

**Issue by bzbarsky-apple** [project-chip/connectedhomeip#16035]
#### Problem
`FabricInfo::CommitToStorage` has code to handle `mOperationalKey == nullptr`, which is not a usable state.  We should not be storing those.

 #### Proposed Solution
Stop storing them.

**Comment by bzbarsky-apple** [comment]
No longer relevant after #19277 because now we can in fact have a null operational key, as long as the keystore has a key.
